src/shell/Session.ts

```diff
@@ -22,7 +22,7 @@ export class Session extends EmitterWithUniqueID {
     history = History;
     historicalPresentDirectoriesStack = new OrderedSet<string>();
 
-    constructor(private application: ApplicationComponent, private _dimensions: Dimensions) {
+    constructor(private application: ApplicationComponent, private _dimensions: Dimensions, public position: Positions) {
```

Comment:
It's not correct to store position on the "model" for two reasons:

1. Position can change during the lifetime of a session. Like if you create three rows, and then delete the second one, the third row's position should change.
2. It's really view-related data.